### PR TITLE
Allows headers to be sent with posted events

### PR DESCRIPTION
**Pull request by dehora**
For #238

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,5 @@ gradle-app.setting
 # gradle/wrapper/gradle-wrapper.properties
 
 target/
+
+*/**/out
```
Comment (adyach) [inline .gitignore]: no new line

**Comment by adyach** [comment]
@dehora do you think the scope of pr can be extended to support headers in other resources?

File: nakadi-java-client/src/test/java/nakadi/StreamProcessorTest.java [hunk 2]
```diff
@@ -390,6 +390,45 @@ public void acceptEncodingGzipIsDefaulted() throws Exception {
...
+        .build();
+
+    StreamConfiguration sc = new StreamConfiguration()
+        .eventTypeName("foo")
+        .batchLimit(2)
+        .streamLimit(5)
+        .streamLimit(1)
+         // tell okhttp to ask the server for unencoded data
+        .requestHeader("Accept-Encoding", "identity")
+        ;
```
Comment (adyach) [inline nakadi-java-client/src/test/java/nakadi/StreamProcessorTest.java]: probably it was unintentional

File: nakadi-java-client/src/test/java/nakadi/StreamProcessorTest.java [hunk 3]
```diff
@@ -390,6 +390,45 @@ public void acceptEncodingGzipIsDefaulted() throws Exception {
...
+    NakadiClient client = NakadiClient.newBuilder()
+        .baseURI(baseURI)
+        .enableHttpLogging()
+        .build();
+
+    StreamConfiguration sc = new StreamConfiguration()
+        .eventTypeName("foo")
+        .batchLimit(2)
+        .streamLimit(5)
+        .streamLimit(1)
```
Comment (adyach) [inline nakadi-java-client/src/test/java/nakadi/StreamProcessorTest.java]: maybe only one call to streamLimit is enough ?

**Comment by adyach** [comment]
LGTM

**Comment by dehora** [comment]
> do you think the scope of pr can be extended to support headers in other resources?

It's already available on consumers (added a test case and updated the readme to show that). Otherwise, yes but I'd rather ship small specific prs and not keep extending this one.

**Comment by dehora** [comment]
Available in 0.9.4

**Pull request merged**